### PR TITLE
Run Canton tests with JDK11

### PR DESCRIPTION
**Pull request by cocreature**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool-on-canton/canton-test-runner.sh [hunk 1]
```diff
@@ -6,8 +6,11 @@ set -e
 set -u
 set -o pipefail
 
+# Canton has some issues with JDK8 so we run it with JDK11.
+JAVA="$(rlocation jdk11_nix/bin/java)"
```
Comment (garyverhaegen-da) [inline ledger/ledger-api-test-tool-on-canton/canton-test-runner.sh:10]: May be worth setting JAVA_HOME. I'm not sure what happens if you run a djk11 binary with JAVA_HOME pointing to jdk8. (Which might be the case here? I'm not sure who/what sets JAVA_HOME for the rest of our build.)
```
export JAVA_HOME=$(realpath $(dirname $(rlocation jdk11_nix/bin/java)/..))
```
should work I think.

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 2]
```diff
@@ -98,6 +99,7 @@ conformance_test(
         "@curl_nix//:bin/curl",
         "@grpcurl_nix//:bin/grpcurl",
         "@jq_dev_env//:jq",
+        "@jdk11_nix//:bin/java",
```
Comment (oliverse-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:102]: on line 94 above we need `s/canton/canton_deploy.jar/` and ideally also add these two changes to the manual test below at lines 130 and 138

**Review by oliverse-da — COMMENTED** [review]
Thank you @moritzkiefer-da for helping with this. Much appreciated 🙇 

With the proposed changes inline, all tests pass on my machine.

File: ledger/ledger-api-test-tool-on-canton/canton-test-runner.sh (hunk 1 above)
Comment (cocreature) [reply ledger/ledger-api-test-tool-on-canton/canton-test-runner.sh:10]: hm, I thought JAVA_HOME doesn’t matter if you invoke the binary directly but maybe I’m wrong.

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel (hunk 2 above)
Comment (cocreature) [reply ledger/ledger-api-test-tool-on-canton/BUILD.bazel:102]: good catch, fixed

**Review by oliverse-da — APPROVED** [review]

File: ledger/ledger-api-test-tool-on-canton/canton-test-runner.sh (hunk 1 above)
Comment (cocreature) [reply ledger/ledger-api-test-tool-on-canton/canton-test-runner.sh:10]: Looking through https://stackoverflow.com/questions/2025290/what-is-java-home-how-does-the-jvm-find-the-javac-path-stored-in-java-home, the JVM itself does not care about it at all. It’s used by other stuff to find java which doesn’t matter if we already invoke java.

Comment (garyverhaegen-da) [reply ledger/ledger-api-test-tool-on-canton/canton-test-runner.sh:10]: It's also sometimes used to find bits of java other than the executable. I'm not concerned about it not being set, as the failures modes for that are well understood (non-zero exit), but I am a little bit concerned about what happens if it is set but to a different JDK than the executable you're running.

It's likely we're not using anything that uses it, though. It's mostly used by "old" programs like Java IDEs and the like.

**Pull request merged**